src/redux/vespaAPI.jsx

```diff
@@ -18,7 +18,18 @@ export const vespaApi = createApi({
     getAllVespas: builder.query({
       query: () => '/rooms',
     }),
+    getAllReservations: builder.query({
+      query: () => '/reservations',
+    }),
+
+    createReservation: builder.mutation({
+      query: (body) => ({
+        url: '/reservations',
+        method: 'POST',
+        body,
+      }),
+    }),
```

Comment:
- Please notice when I click to reserve a book I get a non-authorized error.

![image](https://github.com/Thoth1111/Fullstack-Capstone-Project/assets/96848068/858e3a18-2f3a-4c6e-b019-3b6b36867dbd)

- Please notice here components are rendering more than once which can cause problems in functionality. it would be great if you can use effect structure or any of your preferred ones to make sure the rendering only is happening at the correct number 💯 .